### PR TITLE
Improve PBO handling in WebGL 2

### PR DESCRIPTION
**Pull request by mosra**
In particular:

 * If a PBO gets deleted via `glDeleteBuffer()`, clean up the references to it so the following `glReadPixels()` / `gl(Compressed)Tex(Sub)Image[23]D()` calls are not mistakenly using the buffer overloads, leading to GL errors like "no buffer bound to GL_PIXEL_UNPACK_BUFFER".
 * If a PBO is bound, use it also for `glCompressedTex(Sub)Image[23]D()`. Previously it was used only for the non-compressed calls.

I have quite extensive tests for this in [Magnum](https://magnum.graphics), but couldn't find any matching test case here. If you have a place where a test could be updated, please point me to it :)

The `glTex(Sub)Image[23]D()` has another branch that tests for `pixels != null` ([here](https://github.com/mosra/emscripten/blob/754f0c236b73f4baba9ba78c6ea7a52fc9f0df42/src/library_gl.js#L1621-L1628)), not sure if the same should be done for the compressed variants?

Also not sure if the naming inconsistency (`GLctx.currentPixelPackBufferBinding` vs `GL.currArrayBuffer`) was intentional or accidental. Maybe renaming that to be `GL.currPixelPackBuffer` as well?

### CODE REVIEW

**Comment by kripken** [comment]
Thanks @mosra!

> I have quite extensive tests for this in Magnum, but couldn't find any matching test case here. If you have a place where a test could be updated, please point me to it :)

Good to know about the Magnum testing. I think we don't need 100% extensive testing here, but at minimum we should land this with a test for one of the methods using the binding, and for the clearing of the reference. I don't see a similar current test, so I think we should just add a new one. Can be right after `test_webgl2_packed_types` in `tests/test_browser.py`.

> The glTex(Sub)Image[23]D() has another branch that tests for pixels != null (here), not sure if the same should be done for the compressed variants?

Not sure, perhaps @juj knows?

> Also not sure if the naming inconsistency (GLctx.currentPixelPackBufferBinding vs GL.currArrayBuffer) was intentional or accidental. Maybe renaming that to be GL.currPixelPackBuffer as well?

Heh, yeah, that looks unintentional :) We should rename it to the shorter `curr` for consistency. Can leave that for later.

File: tests/test_browser.py [hunk 1]
```diff
@@ -2528,6 +2528,10 @@ def test_webgl_with_closure(self):
   def test_webgl2_packed_types(self):
     self.btest(path_from_root('tests', 'webgl2_draw_packed_triangle.c'), args=['-lGL', '-s', 'USE_WEBGL2=1', '-s', 'GL_ASSERTIONS=1'], expected='0')
 
+  @requires_graphics_hardware
+  def test_webgl_pbo(self):
```
Comment (kripken) [inline tests/test_browser.py]: `webgl` => `webgl2`

Comment (mosra) [reply tests/test_browser.py]: D'oh! Copypaste bug. Fixed.

**Comment by kripken** [comment]
Great, thanks @mosra!
